webapp/src/timeMachine.tsx

```diff
@@ -145,82 +158,45 @@ export const TimeMachine = (props: TimeMachineProps) => {
         }
     }, [loading, importProject.current, text]);
 
-    const onTimeSelected = (newValue: number) => {
+    React.useEffect(() => {
+        setEntries(getTimelineEntries(history));
+    }, [history]);
+
+    const onTimeSelected = async (newValue: number) => {
         if (importProject.current) {
             setSelected(newValue);
 
             if (newValue === -1) {
                 importProject.current(text);
             }
             else {
-                const previewFiles = applyUntilTimestamp(text, history, newValue);
-                importProject.current(previewFiles)
+                const { files } = await getTextAtTimestampAsync(text, history, newValue);
+                importProject.current(files)
             }
         }
     };
 
     const onGoPressed = React.useCallback(() => {
-        if (selected === -1) {
-            hideDialog();
-        }
-        else {
-            const previewFiles = applyUntilTimestamp(text, history, selected);
-            onProjectLoad(previewFiles, history.find(e => e.timestamp === selected).editorVersion, selected);
-        }
+        (async () => {
```

Comment:
I'm missing why the iife's here? shouldn't 

```typescript
const onGoPressed = React.useCallback(async () => {
```

be fine or is there some typing weirdness from there? Or does `React.useCallback` actually look at returned value and do something if promise-like?

Reply:
oh... i guess that's true

Reply:
i'll fix it in the next pr